file-upload-example/README.md

```diff
@@ -0,0 +1,139 @@
+
+Removing and uploading a file requires a valid user session, we are only focusing on Storage API in this tutorial. Therefor we will use an already created user through the Appwrite console. __Please make sure__ to replace `email` and `password` in the `dummyLogin` function.
+
+#### Uploading a file
+
+```ts
+  public uploadFile(file: File): Observable<object> {
+    return from(this.sdk.storage.createFile(file, ['*'], ['*']));
+  }
+```
+
+We just need to pass through our `File` to upload and a set permissions. The first array is an array for `read` and the second is for `write` permissions. For this tutorial we wil use `*` (wildcard) which means this is available publicly. You can learn more about permissions [here](https://appwrite.io/docs/permissions).
+
+#### Listing the files
+
+```ts
+  public listAllStorageFiles(): Observable<object> {
+    return from(this.sdk.storage.listFiles(null, null, null, null));
+  }
+```
+
+To list files you can pass more parameters for filtering, paging and ordering, but for this tutorial we will list everything.
+
+#### Removing a file
+
+```ts
+  public removeFile(id: string): Observable<object> {
+    return from(this.sdk.storage.deleteFile(id));
+  }
+```
+
+To remove a file, you only need to pass the file's `id`.
+
+#### Previewing a file
+
+```ts
+  public previewFile(id: string): string {
+    return this.sdk.storage.getFilePreview(id, 100, 100, null, 'fff', null);
+  }
+```
+
+To preview a file you'll need the `fileId` and optionally you can also set `width`, `height`, `quality`, `backgroundColor` (hex code without `#`) and `output` to convert your file into your needs. For now we will request a `100×100` preview on a white background.
+
+#### Downloading a file
+
+```ts
+  public downloadFileLink(id: string): string {
+    return this.sdk.storage.getFileDownload(id);
+  }
+```
+
+To get a download link, we'll need the `fileId` and this will return a simple string that we can bind to an `a` tag `href` property.
+
+### Closing words
+
+#### Short app flow
+
+1. App routing
+2. List files routing
+3. Pre-load files through `ListFilesResolver`
+4. Render `ListFilesComponent`
+
+Everything happens under `ListFilesComponent` for simplicity. After creating or removing a file we re-request all the files to show it correctly. 
+Everything else is plain old Angular stuff and wiring together our created `AppwriteService`.
+
+## Congratulations! 🥳 🥳 🥳
+
+Give yourself a pat on the back, since you just successfully created a file upload web app using Appwrite and Angular! You should now have enough knowledge in order to use the Storage API with Appwrite and Angular and if you have any more questions do not hesitate to contact us on [Discord](https://appwrite.io/discord) and if you notice any errors with this tutorial then feel free to raise an Issue or PR in the [GitHub of this tutorial](https://github.com/appwrite/demos-for-angular/).
```

Comment:
```suggestion
Give yourself a pat on the back, since you just successfully created a file upload web app using Appwrite and Angular! You should now have enough knowledge in order to use the Storage API with Appwrite and Angular and if you have any more questions do not hesitate to contact us on [Discord](https://appwrite.io/discord). When you notice any errors with this tutorial, please create an issue or pull request in the [GitHub repository of this tutorial](https://github.com/appwrite/demos-for-angular/).
```